tensorflow_model_optimization/python/core/sparsity_tf2/riglpruner_test.py

```diff
@@ -0,0 +1,577 @@
+    weight = tf.Variable(np.linspace(1.0, 100.0, 100))
+    weight_dtype = weight.dtype.base_dtype
+    mask = tf.Variable(
+        tf.ones(weight.get_shape(), dtype=weight_dtype),
+        dtype=weight_dtype)
+    sparse_vars = [(mask, weight, self.grad(weight))]
+    drop_ratio = 0.
+
+    p = pruner.RiGLPruner(
+      update_schedule=self.updater(drop_ratio, begin_step, end_step, freq_update),
+      sparsity=self.target_sparsity,
+      block_size=self.block_size,
+      block_pooling_type=self.block_pooling_type,
+      seed=self.seed,
+      noise_std=self.noise_std,
+      reinit=self.reinit
+    )
+
+    optimizer = tf.keras.optimizers.SGD(learning_rate=0.01)
+    optimizer.iterations.assign(0)
+    p.create_slots(optimizer, weight)
+
+    for _ in tf.range(end_step + 2):
+      mask_before_update = optimizer.get_slot(weight, 'mask').read_value()
+      before_sparsity = np.count_nonzero(mask_before_update) / tf.size(mask_before_update)
+      step = optimizer.iterations
+      p.update_masks(sparse_vars, step)
+      mask_after_update = optimizer.get_slot(weight, 'mask').read_value()
+      after_sparsity = np.count_nonzero(mask_after_update) / tf.size(mask_before_update)
+      self.assertAllEqual(mask_before_update, mask_after_update)
+
+  @parameterized.parameters(
+    (0.9,),  (0.7,), (0.5,), (0.3,), (0.1,)
+  )
+  def testMaskInitializedCorrectly(self, target_sparsity):
+    weight = tf.Variable(np.linspace(1.0, 100.0, 10), dtype=tf.float32)
+    weight_dtype = weight.dtype.base_dtype
+    mask = tf.Variable(
+        tf.ones(weight.get_shape(), dtype=weight_dtype),
+        dtype=weight_dtype)
+    grad = tf.constant([67.,  45., 89.,  56., 100., 34., 1., 23., 12., 78.])
+    sparsity_params = {
+      'pruning_schedule': self.updater(0.5, 1, 4, 2)
+    } # dummy just for using the pruning optimizer
+    sparsity_config = pruning_config.LowMagnitudePruningConfig(**sparsity_params) # dummy
+
+    p = pruner.RiGLPruner(
+      update_schedule=self.updater(0.5, 1, 4, 2), # update iter 1 and 3
+      sparsity=target_sparsity,
+      block_size=self.block_size,
+      block_pooling_type=self.block_pooling_type,
+      seed=self.seed,
+      noise_std=self.noise_std,
+      reinit=self.reinit
+    )
+
+    _optimizer = tf.keras.optimizers.SGD(learning_rate=0.01)
+    optimizer = pruning_optimizer.PruningOptimizer(_optimizer, sparsity_config)
+    optimizer.iterations.assign(0)
+
+    p.create_slots(optimizer, weight)
+    # check that the mask has the corect sparsity at initialization
+    mask_before_update = optimizer.get_slot(weight, 'mask').read_value()
+    before_sparsity = tf.math.count_nonzero(mask_before_update, dtype=tf.int32) / tf.size(mask_before_update)
+    self.assertAllEqual(before_sparsity, target_sparsity)
+
+
+  @parameterized.parameters(
+    (0.5,), (0.4),
+  )
+  def testUpdatesMaskCorrectly(self, drop_ratio):
+    # this test walks through a concrete example of how a mask will be updated
```

Comment:
Seems like the explanation is still missing some context? "round up to the same". The same what? 0.4 * 0.5 = 0.2, 0.5 * 0.5 = 0.25. Not clear what's being rounded to what?

Also, can you add a note saying the values are computed by hand? (Maybe give brief instructions about how to recompute them by hand if need be? e.g. if we eventually need to test other configurations like this)